app/Repositories/Client/ProductRepository.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace App\Repositories\Client;
+
+use App\Models\Product;
+use App\Models\Rate;
+use Illuminate\Support\Facades\Auth;
+
+class ProductRepository implements ProductRepositoryInterface
+{
+    /**
+     * @var Product
+     */
+    protected $product;
+    protected $rate;
+
+    public function __construct(Product $product, Rate $rate)
+    {
+        $this->product = $product;
+        $this->rate = $rate;
+    }
+
+
+    public function find($id)
+    {
+        try {
+            $product = $this->product->findOrFail($id);
+        } catch (\Exception $exception) {
+            return null;
+        }
+        return $product;
+    }
+
+    public function new()
+    {
+        $product = Product::all()->sortByDesc('created_at');
+        return $product;
+    }
+
+    public function hot()
+    {
+        $rate = $this->rate->get()->sortByDesc('rating');
+        $collection = array();
+        foreach ($rate as $r) {
+            array_push($collection, $this->product->all()->where('id',$r->product_id));
```

Comment:
dòng này e định xử lý gì đó?

Reply:
em định lấy những sản phẩm có lượt rate cao nhất ạ.

Reply:
$this->product->all() đoạn này để lấy tất cả sản phẩm à e

Reply:
khoảng trắng sau dấu , nhé